public/components/agent_framework_traces_flyout_body.test.tsx

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import '@testing-library/jest-dom/extend-expect';
+import { act, waitFor, render, screen, fireEvent } from '@testing-library/react';
+import * as chatContextExports from '../contexts/chat_context';
+import { AgentFrameworkTracesFlyoutBody } from './agent_framework_traces_flyout_body';
+
+jest.mock('./agent_framework_traces', () => {
+  return {
+    AgentFrameworkTraces: () => <div />,
+  };
+});
+
+describe('<AgentFrameworkTracesFlyout/> spec', () => {
+  it('show back button if traceId exists', async () => {
+    const onCloseMock = jest.fn();
+    jest.spyOn(chatContextExports, 'useChatContext').mockReturnValue({
+      traceId: 'test-trace-Id',
+      setSelectedTabId: onCloseMock,
+    });
+    render(<AgentFrameworkTracesFlyoutBody />);
+    expect(screen.queryAllByLabelText('back')).toHaveLength(1);
+    act(() => {
+      fireEvent.click(screen.getByText('Back'));
+    });
+    await waitFor(() => {
+      expect(onCloseMock).toHaveBeenCalledWith('chat');
+    });
+  });
+
+  it('not back button if traceId does not exist', async () => {
+    jest.spyOn(chatContextExports, 'useChatContext').mockReturnValue({
+      traceId: undefined,
+    });
+    render(<AgentFrameworkTracesFlyoutBody />);
+    expect(screen.queryAllByLabelText('back')).toHaveLength(0);
+  });
+
+  it('fullscreen with history open', async () => {
```

Comment:
Shall we update the test description? Seems the history is not open here. We may need to add a separate test for history open.

Reply:
Thanks, I've added a new test for it.